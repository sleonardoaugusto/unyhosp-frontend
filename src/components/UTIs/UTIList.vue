<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(u, i) in utis"
      :key="i"
      class="card col-md-3"
      :to="{name: 'beds', params: {utiId: u.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ u.name }}</p>
      </div>
    </router-link>
  </div>
</template>

<script>
import UTIService from '@/services/UTIService'

export default {
  name: 'UTIList',
  data: () => ({
    utis: []
  }),
  created() {
    const id = this.$route.params.hospitalId
    UTIService.getByHospitalId(id)
      .then((res) => {
        this.utis = res.data
      })
  }
}
</script>

<style scoped>

</style>