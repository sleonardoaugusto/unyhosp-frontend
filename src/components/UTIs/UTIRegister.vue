<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <h2 class="type-h4 type-blue-smoke">Cadastrar Hospital</h2>
    </div>
    <form @submit.prevent="submit" class="col-md-6">
      <div class="form-group" :class="{ 'form-group--error': $v.uti.name.$error }">
        <label class="form__label" for="UTIName">Nome:</label>
        <input class="form__input" type="text" id="UTIName" v-model.trim="$v.uti.name.$model">
        <div v-if="!$v.uti.name.required" class="error--message">Campo obrigat√≥rio</div>
      </div>
      <button class="button button-green" type="submit">CADASTRAR</button>
    </form>
  </div>
</template>

<script>
import UTIService from '@/services/UTIService'
import { required } from 'vuelidate/lib/validators'

export default {
  name: 'HospitalRegister',
  props: ['hospitalId'],
  data: () => ({
    uti: {
      name: ''
    }
  }),
  validations: {
    uti: {
      name: { required }
    }
  },
  methods: {
    submit() {
      this.$v.$touch()
      if (!this.$v.$invalid) {
        const data = {
          name: this.uti.name,
          hospital: this.hospitalId
        }
        UTIService.post(data)
      }
    }
  }
}
</script>

<style scoped>

</style>