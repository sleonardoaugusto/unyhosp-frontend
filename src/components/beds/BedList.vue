<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import BedCard from '@/components/beds/BedCard'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    /* TODO
    *   Get beds*/
  }
}
</script>

<style scoped>

</style>