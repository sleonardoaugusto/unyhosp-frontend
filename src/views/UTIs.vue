<template>
  <div class="col-md-12">
    <div class="row">
      <div class="col-md-12 d-flex justify-content-center">
        <h1 class="type-h3 type-blue-smoke">UTIs</h1>
      </div>
    </div>
    <div class="row">
      <u-t-i-list/>
    </div>
  </div>
</template>

<script>
import UTIList from '@/components/UTIs/UTIList'

export default {
  name: 'UTI',
  components: {
    UTIList
  }
}
</script>

<style scoped>

</style>