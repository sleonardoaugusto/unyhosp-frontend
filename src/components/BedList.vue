<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import beds from '@/mocks/beds'
import BedCard from '@/components/BedCard'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    this.beds = beds.data
  }
}
</script>

<style scoped>

</style>