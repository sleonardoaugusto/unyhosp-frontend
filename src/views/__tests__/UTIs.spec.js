import { shallowMount } from '@vue/test-utils'
import UTIs from '@/views/UTIs'
import BedList from '@/components/BedList'

describe('UTIs.vue', () => {
  let wrapper

  wrapper = shallowMount(UTIs)

  it('renders component properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders child component', () => {
    expect(wrapper.find(BedList)).toBeTruthy()
  })
})