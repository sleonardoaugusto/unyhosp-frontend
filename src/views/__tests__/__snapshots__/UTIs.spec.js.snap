// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UTIs.vue renders component properly 1`] = `
<div class="col-md-12">
  <div class="row">
    <div class="col-md-12 d-flex justify-content-center">
      <h1 class="type-h3 type-blue-smoke">UTIs</h1>
    </div>
  </div>
  <div class="row">
    <u-t-i-list-stub></u-t-i-list-stub>
  </div>
</div>
`;
